src/mlpack/core/data/load_impl.hpp

```diff
@@ -90,7 +92,7 @@ bool Load(const std::string& filename,
           arma::Mat<eT>& matrix,
```

Comment:
```suggestion
          MatType& matrix,
```
I would suggest also to apply this change over the entire file, in addition to the implementation and to the save functions.

Reply:
How should I handle `arma::Col<eT>& vec`

Should I replace it with `MatType` as well?

Reply:
Yes, you can use `MatType` as well.
However, in this case, we need to use `SFINAE` in order to know which overload to chose if the `MatType` is a matrix or a column. Here is an example, but I am not sure of the syntax:
```
Load(filename, 
     const typename std::enable_if<std::is_same<MatType, arma::col>>::type* = 0,
     bool)
```
The idea is to activate this overload if the type of the `MatType` is equal to `arma::col`

Reply:
You need to add a similar one, if this was `col` or `vec` I think, we can do the same for the images type, but let us keep the load image function until the end.
I think when this compiles, we can start removing `boost::spirit`.

Reply:
Okay, sounds good...
We have to handle the `undefined reference` error and get this much running and after that I will start using `SFINAE`